### PR TITLE
Task_8

### PR DESCRIPTION
**Pull request by vitalijkas**

### CODE REVIEW

File: force-app/main/default/classes/PDFContractFromOppController.cls [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                seller_company = usr.Name;
+                labelFrontPageFormattedText = System.Label.pdfFrontPageText.replace('{custmer}', customer).replace('{seller company}', seller_company);
+            }
+        }
+        sumOfQuantity = 0;
+        sumOfPrice = 0;
+        for(OpportunityLineItem p : products){
+            sumOfQuantity += p.Quantity;
+            sumOfPrice += p.UnitPrice;
+        }
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppController.cls:30]: Something wrong with aligment

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@isTest
+private class PDFContractFromOppControllerTest {
+    private static final Integer OBJECT_NUMBER = 1;
+
+    @TestSetup static void testSetup(){
+        List<Opportunity> testOpportunities = TestDataFactory.createOpportunities(OBJECT_NUMBER);
+        insert testOpportunities;
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:7]: Suggestions to add empty lines after insert statement

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @isTest static void testController(){
+        Test.StartTest(); 
+            List<Opportunity> opp = [SELECT Id, Name, ContractId FROM Opportunity LIMIT 1];
+            ApexPages.StandardController sc = new ApexPages.StandardController(opp.get(0));
+            PageReference pageRef = Page.PDFContractFromOpp;
+            PDFContractFromOppController controller = new PDFContractFromOppController(sc);
+            Test.setCurrentPage(pageRef);
+        Test.StopTest();
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:31]: Asserts are required

File: force-app/main/default/staticresources/pdfCss.css [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+a:link{
+    text-decoration: none;
+    color: black;
+}
+th, td{
+    text-align: center;
+}
+
+
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/staticresources/pdfCss.css:35]: remove one line from here

File: force-app/main/default/classes/PDFContractFromOppController.cls [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public Opportunity opp{get; set;}
+    public List<OpportunityLineItem> products{get; set;}
+    public String customer{get; set;}
+    public String seller_company{get; set;}
+    public Decimal sumOfQuantity{get; set;}
+    public Decimal sumOfPrice{get; set;}
+    public String labelFrontPageFormattedText{get; set;}
+
+	public PDFContractFromOppController(ApexPages.StandardController controller) {
+		opp = (Opportunity)controller.getRecord();
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppController.cls:11]: Investigating

File: force-app/main/default/classes/TestDataFactory.cls [hunk 6]
```diff
@@ -13,6 +13,35 @@ public class TestDataFactory {
...
+            cont.FirstName = 'OkAccount' + i;
+            cont.LastName = 'OkLast' + i;
+            testContacts.add(cont);
+        }
+        return testContacts;
+    }
+
+    public static List<User> createUsers(Integer numOfUsers, Id profileId){
+        List<User> testUsers = new List<User>();
+        for(Integer i=0; i<numOfUsers; i++){
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/TestDataFactory.cls:29]: Wrong aligment

File: force-app/main/default/classes/TestDataFactory.cls [hunk 7]
```diff
@@ -13,6 +13,35 @@ public class TestDataFactory {
...
+            User u = new User();
+            u.Alias = 'standt';
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/TestDataFactory.cls:31]: Suggest to use ``userRecord`` except ``u``

File: force-app/main/default/classes/PDFContractFromOppController.cls (hunk 5 above)
Comment (vitalijkas) [reply force-app/main/default/classes/PDFContractFromOppController.cls:11]: Illegal assignment from SObject to Opportunity (11:3)

File: force-app/main/default/classes/TestDataFactory.cls [hunk 8]
```diff
@@ -13,6 +13,35 @@ public class TestDataFactory {
...
+            cont.LastName = 'OkLast' + i;
+            testContacts.add(cont);
+        }
+        return testContacts;
+    }
+
+    public static List<User> createUsers(Integer numOfUsers, Id profileId){
+        List<User> testUsers = new List<User>();
+        for(Integer i=0; i<numOfUsers; i++){
+            User user = new User();
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/TestDataFactory.cls:30]: User - it's bad name
Possible conflicts  !!!

Use ``userRecord``

**Comment by BohuslavskiyVolodymyrRedtag** [comment]
You still have issues with alignment

File: force-app/main/default/classes/PDFContractFromOppController.cls [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    public Decimal sumOfQuantity{get; set;}
+    public Decimal sumOfPrice{get; set;}
+    public String labelFrontPageFormattedText{get; set;}
+
+	public PDFContractFromOppController(ApexPages.StandardController controller) {
+		opportunityRecord = (Opportunity)controller.getRecord();
+        opportunityRecord = [SELECT Id, ContractId FROM Opportunity WHERE Id =: opportunityRecord.Id];
+        products = [SELECT Product2Id, Quantity, UnitPrice FROM OpportunityLineItem WHERE OpportunityId =: opportunityRecord.Id];
+
+        if(opportunityRecord.ContractId != null){
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppController.cls:15]: Please use the String.isNotBlanck();

File: force-app/main/default/classes/PDFContractFromOppController.cls [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	public Opportunity opportunityRecord { get; set; }
+	public List<OpportunityLineItem> products { get; set; }
+	public String customer { get; set; }
+	public String seller_company { get; set; }
+	public Decimal sumOfQuantity { get; set; }
+	public Decimal sumOfPrice { get; set; }
+	public String labelFrontPageFormattedText { get; set; }
+
+	public PDFContractFromOppController(ApexPages.StandardController controller) {
+		opportunityRecord = (Opportunity) controller.getRecord();
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppController.cls:11]: Suggest to use ``controller.getId();`` and hardcast dont needed;
And use subQuerys

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        testContracts.get(0).CustomerSignedId = testContacts.get(0).Id;
+        update testContracts;
+        
+        testOpportunities.get(0).ContractId = testContracts.get(0).Id;
+        update testOpportunities;
+    }
+
+    @isTest static void testController(){
+        Test.StartTest(); 
+            List<Opportunity> opp = [SELECT Id, ContractId FROM Opportunity LIMIT 1];
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:32]: Use ``LIMIT 1`` statement, without ``.get(0)``

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+            ApexPages.StandardController sc = new ApexPages.StandardController(opp.get(0));
+            PageReference pageRef = Page.PDFContractFromOpp;
+            PDFContractFromOppController controller = new PDFContractFromOppController(sc);
+            Test.setCurrentPage(pageRef);
+        Test.StopTest();
+        System.assertEquals(controller.opp, opp.get(0));
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:38]: In this case, I think to be better if you’ll check ``pageRef != Null``

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+    @isTest static void testController(){
+        Test.StartTest(); 
+            Opportunity testOpportunity = [SELECT Id, ContractId FROM Opportunity LIMIT 1];
+            ApexPages.StandardController sc = new ApexPages.StandardController(testOpportunity);
+            PageReference pageRef = Page.PDFContractFromOpp;
+            PDFContractFromOppController controller = new PDFContractFromOppController(sc);
+            Test.setCurrentPage(pageRef);
+        Test.StopTest();
+        System.assert(pageRef != null);
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:38]: Suggestion use ``asertNotEquals`` statement

File: force-app/main/default/classes/PDFContractFromOppControllerTest.cls [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+    @isTest static void testController(){
+        Test.StartTest(); 
+            Opportunity testOpportunity = [SELECT Id, ContractId FROM Opportunity LIMIT 1];
+            ApexPages.StandardController sc = new ApexPages.StandardController(testOpportunity);
+            PageReference pageRef = Page.PDFContractFromOpp;
+            PDFContractFromOppController controller = new PDFContractFromOppController(sc);
+            Test.setCurrentPage(pageRef);
+        Test.StopTest();
+        System.assertNotEquals(pageRef, null);
```
Comment (BohuslavskiyVolodymyrRedtag) [inline force-app/main/default/classes/PDFContractFromOppControllerTest.cls:38]: replace

**Comment by BohuslavskiyVolodymyrRedtag** [comment]
Resolve conflicts and feel free to marge it

**Pull request merged**